mysql-test/suite/innodb/t/import_hidden_fts.test

```diff
@@ -0,0 +1,46 @@
+--source include/have_innodb.inc
+--source include/have_sequence.inc
+# Table with virtual, fulltext, instant add, instant drop column
+# and purgeable rows
+call mtr.add_suppression("InnoDB: Added system generated FTS_DOC_ID and FTS_DOC_ID_INDEX while importing the tablespace");
+CREATE TABLE t1(f1 INT NOT NULL PRIMARY KEY,
+		f2 CHAR(2) not null, fulltext f_idx(f2),
+		f3 INT as (f1) VIRTUAL, INDEX(f3),
+		f4 INT as (f1) STORED, INDEX(f4),
+		f5 INT as (f1) VIRTUAL)ENGINE=InnoDB;
+INSERT INTO t1(f1, f2) VALUES(1, "on");
+INSERT INTO t1(f1, f2) SELECT seq, "si" FROM seq_2_to_16384;
+ALTER TABLE t1 ADD COLUMN f6 INT NOT NULL;
+ALTER TABLE t1 DROP COLUMN f6;
+ALTER TABLE t1 DROP INDEX f_idx;
+connect(con1,localhost,root,,);
+START TRANSACTION WITH CONSISTENT SNAPSHOT;
+
+connection default;
+DELETE FROM t1 WHERE f1 > 1;
+FLUSH TABLE t1 FOR EXPORT;
+let MYSQLD_DATADIR =`SELECT @@datadir`;
```

Comment:
This test hung in an `./mtr --embedded` test [somewhere](https://buildbot.mariadb.org/#/builders/236/builds/19097/steps/6/logs/stdio) after the `FLUSH TABLE t1 FOR EXPORT`:
```
innodb.import_hidden_fts 'innodb'        w11 [ fail ]  timeout after 900 seconds
        Test ended at 2024-02-15 21:58:19
Test case timeout after 900 seconds
```
Because there are no stack traces of the right process, I ran this test on my own system:
```sh
./mtr --embedded innodb.import_hidden_fts
```
The test is taking a very long time on my local system as well. I would suggest the following addition to disable the row-level undo logging, because we should not really need it here:
```diff
diff --git a/mysql-test/suite/innodb/t/import_hidden_fts.test b/mysql-test/suite/innodb/t/import_hidden_fts.test
index 39bb6c62ca7..2cdd397b08c 100644
--- a/mysql-test/suite/innodb/t/import_hidden_fts.test
+++ b/mysql-test/suite/innodb/t/import_hidden_fts.test
@@ -8,8 +8,11 @@ CREATE TABLE t1(f1 INT NOT NULL PRIMARY KEY,
 		f3 INT as (f1) VIRTUAL, INDEX(f3),
 		f4 INT as (f1) STORED, INDEX(f4),
 		f5 INT as (f1) VIRTUAL)ENGINE=InnoDB;
-INSERT INTO t1(f1, f2) VALUES(1, "on");
-INSERT INTO t1(f1, f2) SELECT seq, "si" FROM seq_2_to_16384;
+SET STATEMENT unique_checks=0,foreign_key_checks=0 FOR
+START TRANSACTION;
+INSERT INTO t1(f1, f2) VALUES(1, 'on');
+INSERT INTO t1(f1, f2) SELECT seq, 'si' FROM seq_2_to_16384;
+COMMIT;
 ALTER TABLE t1 ADD COLUMN f6 INT NOT NULL;
 ALTER TABLE t1 DROP COLUMN f6;
 ALTER TABLE t1 DROP INDEX f_idx;
```
With this fix, the test would complete in 13.9 seconds instead of 14.5 seconds on my system. That is still a lot. What do we need to insert that many rows for? If I create a table with 16 rows instead of 16384, the test will complete in 0.13 seconds.

I was curious where the time is being spent. At least `rbt_check_ordering()` seems to be popular, and `fts_doc_id_cmp()` is there as well.